src/components/Explorer.js

```diff
@@ -48,93 +51,131 @@ const mungeSummaryData = (summaries, place) => {
   }))
 }
 
-const Explorer = ({ appState, dispatch, params, router }) => {
-  const crime = lowerCase(params.crime)
-  const { filters, nibrs, summaries } = appState
-  const place = startCase(params.place)
-
-  // show not found page if crime or place unfamiliar
-  if (!crimeSlugs.includes(crime) || !lookup(place)) return <NotFound />
-
-  const nibrsData = filterNibrsData(nibrs.data, filters)
-  const trendData = mungeSummaryData(summaries, params.place)
-
-  return (
-    <div className='site-wrapper'>
-      <Sidebar
-        dispatch={dispatch}
-        filters={filters}
-        router={router}
-      />
-      <div className='site-content'>
-        <div className='container-main mx-auto p3'>
-          <Breadcrumbs {...params} />
-          <div className='md-flex items-baseline mb4 border-bottom border-blue-lighter'>
-            <h1 className='flex-auto mt0 mb1'>
-              {place}, {filters.timeFrom}-{filters.timeTo}
-            </h1>
-          </div>
-          <p className='mb5 fs1 serif'>
-            Incidents of
-            <Term
-              dispatch={dispatch}
-              id={crimeIds[params.crime] || 'undefinedTerm'}
-            >
-              {crime}
-            </Term>
-            are on the
-            rise in {place}, but lower than 5 or 10 years ago.
-            {place}&#39;s {crime} rate surpassed that of the U.S. in 1985, and peaked
-            in 1991, with a rate of over 52 incidents per 100,000
-            people.<sup>1</sup>
-          </p>
-          <hr className='mt0 mb3' />
-          <div className='mb2 p2 sm-p4 bg-blue-lighter'>
-            <div className='md-flex items-baseline'>
-              <div className='flex-auto'>
-                <div className='inline-block'>
-                  <h2 className='m0 fs-ch1 sans-serif'>
-                    Reported {plural(crime)} in {place},
-                    <br />
-                    {filters.timeFrom}–{filters.timeTo}
-                  </h2>
+class Explorer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.props = props
+    this.handleSidebarChange = ::this.handleSidebarChange
+  }
+
+  componentDidMount() {
+    const { appState, dispatch } = this.props
+    const filters = {
+      ...this.props.filters,
+      ...this.props.router.params,
+      ...this.props.router.location.query,
+    }
+
+    const check = value => {
+      const year = parseInt(value, 10)
+      return year >= 1960 && year <= 2014
+    }
+
+    if (!check(filters.timeFrom)) filters.timeFrom = appState.filters.timeFrom
+    if (!check(filters.timeTo)) filters.timeTo = appState.filters.timeTo
+
+    dispatch(updateFilters(filters))
+    if (filters.crime) {
+      dispatch(fetchSummaries(filters))
+      dispatch(fetchNibrsDimensions(filters))
+    }
+  }
+
+  handleSidebarChange(change) {
```

Comment:
Good catch, yup, actually the sidebar doesn't work as of now. Let me pass the function all the way through